main.tf

```diff
@@ -3,61 +3,52 @@ resource "aws_cognito_user_pool" "this" {
 
   sms_authentication_message = var.auth_sms_message
   email_verification_message = var.html_email_message == "" ? var.auth_sms_message : var.html_email_message
-  email_verification_subject = "Código de verificación"
-  mfa_configuration          = "OPTIONAL"
+  email_verification_subject = var.email_verification_subject
 
-  admin_create_user_config {
-    allow_admin_create_user_only = false
-  }
-
-  software_token_mfa_configuration {
-    enabled = true
-  }
```

Comment:
Esto debe ser opcional

Reply:
I added the enable_software_token_mfa variable and the dynamic "software_token_mfa_configuration" block to make the MFA configuration optional.